experimental/FTheoryTools/src/types.jl

```diff
@@ -0,0 +1,107 @@
+################################################
+# 1: The Julia types for FTheoryTools
+################################################
+
+abstract type AbstractFTheoryModel end
+
+ToricOrNonToricCoveredScheme = Union{ToricCoveredScheme{BRT}, CoveredScheme{BRT}} where {BRT<:Ring}
+
+@attributes mutable struct ClosedSubschemeModel <: AbstractFTheoryModel
+  base_space::ToricOrNonToricCoveredScheme
+  ambient_space::ToricOrNonToricCoveredScheme
+  #@req typeof(base_space) === typeof(ambient_space) "Base and ambient space must be of the same type"
+  # total_space
+  # fiber_ambient_space
+  # singular_loci
+  # chow_ring
+  # is_cy
+  # generic_fiber
+  # modell_weil_group
+  # weil_chatelet_group
+  # gauge_group
+  # fiber_diagram (cod. 1,2,3)
+  # is_flat
+  # Q_factorial_terminal
+  # toric_Q_factorial_terminal
+  # jacbian_fibration
+  # mirror_dual
+  ClosedSubschemeModel(base_space::ToricOrNonToricCoveredScheme, ambient_space::ToricOrNonToricCoveredScheme) = new(base_space, ambient_space)
+end
+
+
+@attributes mutable struct CompleteIntersectionModel <: AbstractFTheoryModel
+  base_space::ToricOrNonToricCoveredScheme
+  ambient_space::ToricOrNonToricCoveredScheme
+  defining_ideal::MPolyIdeal
+  # zero section
+  CompleteIntersectionModel(base_space::ToricOrNonToricCoveredScheme, ambient_space::ToricOrNonToricCoveredScheme, defining_ideal::MPolyIdeal) = new(base_space, ambient_space, defining_ideal)
+end
+
+
+@attributes mutable struct HypersurfaceModel <: AbstractFTheoryModel
+  base_space::ToricOrNonToricCoveredScheme
+  ambient_space::ToricOrNonToricCoveredScheme
+  hypersurface_equation::MPolyRingElem{QQFieldElem}
+  HypersurfaceModel(base_space::ToricOrNonToricCoveredScheme, ambient_space::ToricOrNonToricCoveredScheme, hypersurface_equation::MPolyRingElem{QQFieldElem}) = new(base_space, ambient_space, hypersurface_equation)
+end
+
+
+@attributes mutable struct GlobalWeierstrassModel
+  weierstrass_f::MPolyRingElem{QQFieldElem}
+  weierstrass_g::MPolyRingElem{QQFieldElem}
+  weierstrass_polynomial::MPolyRingElem{QQFieldElem}
+  base_space::ToricOrNonToricCoveredScheme
+  ambient_space::ToricOrNonToricCoveredScheme
+  fiber_ambient_space::ToricOrNonToricCoveredScheme
+  function GlobalWeierstrassModel(weierstrass_f::MPolyRingElem{QQFieldElem},
+                            weierstrass_g::MPolyRingElem{QQFieldElem},
+                            weierstrass_polynomial::MPolyRingElem{QQFieldElem},
+                            base_space::ToricOrNonToricCoveredScheme,
+                            ambient_space::ToricOrNonToricCoveredScheme)
+    return new(weierstrass_f, weierstrass_g, weierstrass_polynomial, base_space, ambient_space, weighted_projective_space(ToricCoveredScheme, [2,3,1]))
+  end
+end
+
+
+@attributes mutable struct GlobalTateModel
+  tate_a1::MPolyRingElem{QQFieldElem}
+  tate_a2::MPolyRingElem{QQFieldElem}
+  tate_a3::MPolyRingElem{QQFieldElem}
+  tate_a4::MPolyRingElem{QQFieldElem}
+  tate_a6::MPolyRingElem{QQFieldElem}
```

Comment:
Why not something like
```suggestion
  tate_ai::Vector{MPolyRingElem{QQFieldElem}}
```
?

Then the accessor could also be something like
```
tate_section_ai(bla, i::Int)
```
So you would only need one instead of 5.

Reply:
You could probably also use a `Dict{Symbol, MPolyRingElem{QQFieldElem}}` and then assign these to the symbols `:ai`.

Reply:
Thanks for this suggestion, we will go with the vector option and create an accessor as you suggested.

Reply:
Actually, we do not have `a5` but would prefer the `vector`. So we need a vector that has no 5-th entry sort of...

Is most likely not ultra-critical. Please allow us to ignore this for now.